Automattic/jetpack
# Migration plugin: Create/integrate migration progress screen

Fixes https://github.com/Automattic/dotcom-forge/issues/1492
Fixes https://github.com/Automattic/wp-calypso/issues/72365
Fixes https://github.com/Automattic/dotcom-forge/issues/1639
Fixes https://github.com/Automattic/dotcom-forge/issues/1641
Fixes https://github.com/Automattic/dotcom-forge/issues/1642
Fixes https://github.com/Automattic/dotcom-forge/issues/1643

## Proposed changes:

* Installed `@automattic/jetpack-api` dependency
* Added fetchMigrationStatus API method
* Added ToS link above the Get started button
* Created Migration Progress component
* Integrated Migration Progress into the Migration plugin
* Introduced MigrationError component and handled case when import is in progress

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Launch a [JN site with the migration plugin](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack-migration=origin/update/migration-plugin-2&wp-debug-log).
* Check if there is `ToS` link above the `Get Started` button
* Click on the `Get Started` button
* Start import everything
* Check if the migration plugin change it's screen to the Migration Progress (see screenshot below)
* Check if the `Check your migration progress` button redirects to the proper `migrationHandler` route


## Screenshots
<img width="1131" alt="Screenshot 2023-02-07 at 10 39 11" src="https://user-images.githubusercontent.com/1241413/217517110-d74bc627-c4e4-4fd1-a561-0c35acee9679.png">

![error-component](https://user-images.githubusercontent.com/4074459/218296038-5877ee86-9505-4f7f-8fcc-3668c4d64190.png)

<img width="505" alt="Screenshot 2023-02-09 at 15 54 33" src="https://user-images.githubusercontent.com/1241413/217847646-6fb36534-c353-4d9b-b0ee-58a24f876a53.png">

<img width="370" alt="Screenshot 2023-02-10 at 23 46 54" src="https://user-images.githubusercontent.com/1241413/218215102-12fce2fc-6cfc-437d-8b3b-09685c5bd8fc.png">
